parser/lexer.go

```diff
@@ -204,6 +204,15 @@ func (s *Scanner) scan() (tok int, pos Pos, lit string) {
 		ch0 = s.skipWhitespace()
 	}
 	pos = s.r.pos()
+
+	if ch0 == '\\' {
+		// support \N as shortcut for NULL
+		if s.r.charByIndex(pos.Offset+1) == 'N' {
```

Comment:
Before using `s.r.peek()`，`s.r inc()` must be called first,  because current offset is pointed to '\\'.  

 I want to know if next element is 'N', and don't wanna change offset at the same time .

 If next element is not 'N', and `s.r.inc()` + `s.r.peek()` are called,  we must back to previous offset, it is not convenient .